crates/sui-graphql-rpc/src/types/display.rs

```diff
@@ -3,8 +3,160 @@
+        }
+    }
+}
+
+pub(crate) fn get_rendered_fields(
+    fields: VecMap<String, String>,
+    move_struct: &MoveStruct,
+) -> Result<Vec<DisplayEntry>, Error> {
+    let mut rendered_fields: Vec<DisplayEntry> = vec![];
+
+    for entry in fields.contents.iter() {
+        let rendered_value = match parse_template(&entry.value, move_struct) {
+            Ok(value) => DisplayEntry::create_value(entry.key.clone(), value),
+            Err(e) => DisplayEntry::create_error(entry.key.clone(), e.to_string()),
+        };
+        rendered_fields.push(rendered_value);
+    }
+
+    Ok(rendered_fields)
+}
+
+/// Handles the PART of the grammar, defined as:
+/// PART   ::= '{' CHAIN '}'
+///          | '\{' | '\}'
+///          | [:utf8:]
+/// Defers resolution down to the IDENT to get_value_from_move_struct,
+/// and substitutes the result into the PART template.
+fn parse_template(template: &str, move_struct: &MoveStruct) -> Result<String, DisplayRenderError> {
+    let mut output = template.to_string();
+    let mut var_name = String::new();
+    let mut in_braces = false;
+    let mut escaped = false;
+
+    for ch in template.chars() {
+        match ch {
+            '\\' => {
+                escaped = true;
+                continue;
+            }
+            '{' if !escaped => {
+                in_braces = true;
+                var_name.clear();
+            }
+            '}' if !escaped => {
+                in_braces = false;
+                let value = get_value_from_move_struct(move_struct, &var_name)?;
+                output = output.replace(&format!("{{{}}}", var_name), &value.to_string());
+            }
+            _ if !escaped => {
+                if in_braces {
+                    var_name.push(ch);
+                }
+            }
+            _ => {}
+        }
+        escaped = false;
+    }
+
+    Ok(output.replace('\\', ""))
+}
+
+/// Handles the CHAIN and IDENT of the grammar, defined as:
+/// CHAIN  ::= IDENT | CHAIN '.' IDENT
+/// IDENT  ::= /* Move identifier */
+pub(crate) fn get_value_from_move_struct(
+    move_struct: &MoveStruct,
+    var_name: &str,
+) -> Result<String, DisplayRenderError> {
+    let parts: Vec<&str> = var_name.split('.').collect();
+    if parts.is_empty() {
+        return Err(DisplayRenderError::TemplateValueEmpty);
+    }
+    // todo: 10 is a carry-over from the sui-json-rpc implementation
+    // we should introduce this as a new limit on the config
+    if parts.len() > 10 {
+        return Err(DisplayRenderError::ExceedsLookupDepth(parts.len(), 10));
+    }
+
+    // update this as we iterate through the parts
+    let start_value = &MoveValue::Struct(move_struct.clone());
+
+    let result = parts
+        .iter()
+        .try_fold(start_value, |current_value, part| match current_value {
+            MoveValue::Struct(s) => s
+                .fields
+                .iter()
+                .find_map(|(id, value)| {
+                    if id.to_string() == *part {
```

Comment:
You probably don't need to use `to_string()` here, as opposed to `as_str()` or similar (to avoid copying the string).